src/libraries/Native/Unix/System.Native/CMakeLists.txt

```diff
@@ -23,11 +23,13 @@ set(NATIVE_SOURCES
     pal_sysctl.c
 )
 
-if (CMAKE_SYSTEM_NAME STREQUAL Linux)
+if (NOT CMAKE_SYSTEM_NAME STREQUAL Darwin)
     set(NATIVE_SOURCES ${NATIVE_SOURCES} pal_networkchange.c)
 
-    if (!HAVE_LINUX_RTNETLINK_H)
+    if (CMAKE_SYSTEM_NAME STREQUAL Linux AND !HAVE_LINUX_RTNETLINK_H)
```

Comment:
I would probably not mix them and I would leave Linux alone e.g. If Linux must have HAVE_LINUX_RTNETLINK_H.
For other OSes you can for HAVE_RT_MSGHDR and provide implementation or it would not try to compile it. 
If you feel strongly about this, I would add something like if FreeBSD must have HAVE_RT_MSGHDR as separate block.

Reply:
They are logically equal to what we have in master. `if(Linux) if(!HAVE_LINUX_RTNETLINK_H)`, now `if(Linux AND !HAVE_LINUX_RTNETLINK_H)`. Or do you mean separate out the else-case for this validation or `pal_networkchange.c` as well?

Reply:
I was more thinking about separating BSD (and others) from Linux.

Reply:
Thanks. I have addressed the comment. Could you please give it another pass?